pkg/cmdline/env.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cmdline
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+// EnvAppend combines command line and environment var into a single argument
+func EnvAppend(flag *pflag.Flag, envvar string) error {
+	if err := flag.Value.Set(envvar); err != nil {
+		return fmt.Errorf("unable to set %s to environment variable value %s", flag.Name, envvar)
+	}
+
+	flag.Changed = true
+	sylog.Debugf("Updated flag '%s' value to: %s", flag.Name, flag.Value)
+	return nil
+}
+
+// EnvBool sets a bool flag if the CLI option is unset and env var is set
+func EnvBool(flag *pflag.Flag, envvar string) error {
+	if flag.Changed || envvar == "" {
+		return nil
+	}
+
+	if err := flag.Value.Set(envvar); err != nil {
```

Comment:
the reason why this can return an error is because the value `envvar` is not a boolean (1, t, T, true, TRUE, True, 0, f, F, false, FALSE, False). If the value the user is placing in the environment variable is not a boolean, I'm not sure setting the flag to true is a good idea, as this might lead to surprising behavior (e.g. "VAR=no ...")